src/mariadb_udf/config.c

```diff
@@ -0,0 +1,308 @@
+    { "user", 1 }, // User-level messages
+    { "uucp", 8 }, // UUCP subsystem
+};
+
+int priority_lookup(char* name, ssize_t len)
+{
+    return _lookup(name, len, PRIORITY_LOOKUP, sizeof(PRIORITY_LOOKUP) / sizeof(PRIORITY_LOOKUP[0]));
+}
+
+int facility_lookup(char* name, ssize_t len)
+{
+    return _lookup(name, len, FACILITY_LOOKUP, sizeof(FACILITY_LOOKUP) / sizeof(FACILITY_LOOKUP[0]));
+}
+
+#define get_val(f, v, l)               \
+    do {                               \
+        char* tmp;                     \
+        char* field = strstr(line, f); \
+        v = NULL;                      \
+        if (field == NULL) {           \
+            break;                     \
+        }                              \
+        v = strchr(field, '=');        \
+        v++;                           \
+        tmp = strchr(v, ' ');          \
+        if (tmp == NULL) {             \
+            tmp = end;                 \
+        }                              \
+        val_len = tmp - v;             \
+    } while (0)
+
+int loadconfig(char* path, int* count, struct configuration** out)
+{
+    FILE* fp;
+    char host[256];
+    char port[256];
+    char* line = NULL, *end;
+    size_t len;
+    ssize_t read;
+    struct configuration* configurations = NULL;
+    ;
+    int conf_len = 0;
+    int rc = 0;
+
+    fp = fopen(path, "r");
+    if (fp == NULL) {
+        return errno;
+    }
+
+    while ((read = getline(&line, &len, fp)) != -1) {
+        char* value;
+        ssize_t val_len;
+        struct configuration conf;
+        memset(&conf, 0, sizeof(struct configuration));
+        if (line[0] == '#' || len < 3) {
+            continue;
+        }
+
+        // Chomp off the end
+        if (line[read - 1] == '\n') {
+            read--;
+        }
+
+        end = line + read;
+        get_val("type=", value, val_len);
+        // Check for a valid type
+        if (value == NULL || strncmp("syslog", value, val_len) != 0) {
+            continue;
+        }
+
+        rc = setup_syslog_header(&conf.syslog_header, line, read);
+        if (rc != 0) {
+            continue;
+        }
+
+        rc = setup_cef_header(&conf.cef_header, line, read);
+        if (rc != 0) {
+            continue;
+        }
+
+        get_val("host=", value, val_len);
+        if (value == NULL) {
+            cleanup_syslog_header(&conf.syslog_header);
+            cleanup_cef_header(&conf.cef_header);
+            continue;
+        }
+
+        strncpy(host, value, val_len);
+        host[val_len] = 0;
+
+        get_val("port=", value, val_len);
+        if (value == NULL) {
+            cleanup_syslog_header(&conf.syslog_header);
+            cleanup_cef_header(&conf.cef_header);
+            continue;
+        }
+
+        strncpy(port, value, val_len);
+        port[val_len] = 0;
+        rc = get_udp_socket(host, port, &conf.syslog_fd, (struct sockaddr_storage*)&conf.saddr, &conf.saddr_len);
```

Comment:
close() is never called on conf.syslog_fd

Reply:
Since there is no cleanup routine I never close the socket.
Also If I close it that also means I would need to open it back up.
Since it is an unconnected UDP socket I could just leave it open without any have to call socket() again.
Also having multiple threads share the UDP socket is safe since sending is "atomic".

Reply:
Ok then, so document it in the code please.